ipapython/certdb.py

```diff
@@ -106,13 +114,63 @@ def run_certutil(self, args, stdin=None, **kwargs):
         new_args = new_args + args
         return ipautil.run(new_args, stdin, **kwargs)
 
-    def create_db(self, password_filename):
+    def create_db(self, password_filename=None, user=None, group=None,
+                  mode=None, backup=False):
         """Create cert DB
 
         :param password_filename: Name of file containing the database password
+        :param user: User owner the secdir
+        :param group: Group owner of the secdir
+        :param mode: Mode of the secdir
+        :param backup: Backup the sedir files
         """
+        dirmode = 0o750
+        filemode = 0o640
+        if mode is not None:
+            dirmode = mode
+            filemode = mode & 0o666
+
+        uid = -1
+        gid = -1
+        if user is not None:
+            uid = pwd.getpwnam(user).pw_uid
+        if group is not None:
+            gid = grp.getgrnam(group).gr_gid
+
+        if backup:
+            for filename in NSS_FILES:
+                path = os.path.join(self.secdir, filename)
+                ipautil.backup_file(path)
+
+        if not os.path.exists(self.secdir):
+            os.makedirs(self.secdir, dirmode)
+
+        if password_filename is None:
+            password_filename = self.pwd_file
+
+        if not os.path.exists(password_filename):
+            # Create the password file for this db
+            hex_str = binascii.hexlify(os.urandom(10))
+            with io.open(os.open(password_filename,
```

Comment:
IIRC os.write() only accept a unicode argument in this case but hex_str is a binary string.